subprojects/language-java/src/main/java/org/gradle/api/internal/tasks/compile/incremental/recomp/GroovyRecompilationSpecProvider.java

```diff
@@ -110,6 +111,11 @@ private void processOtherChanges(PreviousCompilation previous, RecompilationSpec
             }
 
             File changedFile = fileChange.getFile();
+            if (FileUtils.hasExtension(changedFile, ".java")) {
```

Comment:
Can we do the inverse? Like `!FileUtils.hasExtension(changedFile, ".groovy")`? What about lower vs. uppercase?

Reply:
Changed to `!FileUtils.hasExtension(changedFile, ".groovy")`, I don't think we need to do special processing to `.GROOVY` files - for the people using `GROOVY` as extension, they shouldn't.